### PR TITLE
GDExtension: Prevent crash during shutdown as singletons are deleted

### PR DESCRIPTION
**Pull request by dsnopek**
This issue was reported by @mihe on [this comment](https://github.com/godotengine/godot-cpp/pull/1458#issuecomment-2104882470)

~~Marking as draft, because I haven't yet managed to reproduce the issue personally, so I don't know if this fixes it.~~

### CODE REVIEW

File: core/object/object.cpp [hunk 1]
```diff
@@ -2095,9 +2095,13 @@ Object::~Object() {
 		_extension_instance = nullptr;
 	}
 #ifdef TOOLS_ENABLED
-	else if (_instance_bindings != nullptr && Engine::get_singleton()->is_extension_reloading_enabled()) {
-		for (uint32_t i = 0; i < _instance_binding_count; i++) {
-			GDExtensionManager::get_singleton()->untrack_instance_binding(_instance_bindings[i].token, this);
+	else if (_instance_bindings != nullptr) {
+		Engine *engine = Engine::get_singleton();
+		GDExtensionManager *gdextension_manager = GDExtensionManager::get_singleton();
+		if (engine && gdextension_manager && engine->is_extension_reloading_enabled()) {
```
[the remark this answers is not in the document]
Comment (mihe) [reply core/object/object.cpp]: You seem to have accidentally removed the actual fix now instead. :)

[the remark this answers is not in the document]
Comment (mihe) [reply core/object/object.cpp]: I can confirm that nulling out the singletons _and_ adding the null checks to `Object::~Object` fixes this issue, and results in a clean exit with Application Verifier.

**Comment by dsnopek** [comment]
@mihe says this fixed it for him above, so I'm taking it out of draft.

**Review by mihe — APPROVED** [review]
LGTM, whatever that's worth. :)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**